File: client/common/protocolo/protocoloApuesta.go
package protocolo

import (
	"net"

	"github.com/7574-sistemas-distribuidos/docker-compose-init/client/common/modelo"
	"github.com/7574-sistemas-distribuidos/docker-compose-init/client/common/serializacion"
)

func EnviarApuestas(conn net.Conn, apuesta []*modelo.Apuesta) (int, error) {
	texto := serializacion.ApuestasAString(apuesta)
	return enviarString(conn, texto)
}

func RecibirApuestas(conn net.Conn) ([]*modelo.Apuesta, error) {
	texto, err := recibirString(conn)
	if err != nil {
		return nil, err
	}

	return serializacion.ApuestasDesdeString(texto), nil
}

File: server/common/protocolo/protocolo_apuesta.py
import socket
from typing import List
from common.utils import Bet
from common.protocolo.protocolo_base import recibir_string
from common.serializacion.serializacion_apuesta import apuestas_a_string, apuestas_desde_string
from common.protocolo.protocolo_base import enviar_string



def recibir_apuestas(socket: socket.socket) -> tuple[List[Bet], int]:
    apuestas_str = recibir_string(socket)
    return apuestas_desde_string(apuestas_str)

def enviar_apuestas(socket: socket.socket, apuestas: List[Bet]):
    texto = apuestas_a_string(apuestas)
    return enviar_string(socket, texto)

